gemma-core/src/main/java/ubic/gemma/core/analysis/preprocess/batcheffects/BatchInfoPopulationServiceImpl.java

```diff
@@ -85,17 +85,8 @@ public static boolean isBatchFactor( ExperimentalFactor ef ) {
         if ( c == null )
             return false;
 
-        boolean isBatchFactor = false;
-
-        boolean looksLikeBatch = ef.getName().equals( ExperimentalDesignUtils.BATCH_FACTOR_NAME );
-
-        if ( c.getCategory() != null && c.getCategory().equals( ExperimentalDesignUtils.BATCH_FACTOR_CATEGORY_NAME ) ) {
-            isBatchFactor = true;
-        } else if ( looksLikeBatch ) {
-            isBatchFactor = true;
-        }
-
-        return isBatchFactor;
+        return ExperimentalDesignUtils.BATCH_FACTOR_CATEGORY_NAME.equals( c.getCategory() )
+                || ExperimentalDesignUtils.BATCH_FACTOR_NAME.equals( ef.getName() );
```

Comment:
This should be added to the 1.30.1. It basically safeguard against null categories or names.